### PR TITLE
Adding Spread operator and Rest parameters to JS section

### PR DESCRIPTION
**Pull request by mserino**
### Description of the Change
Updating the JavaScript documentation with information about Rest parameters and Spread operators, notation available with ES6.

### Benefits
Additional information that can be referenced for specific ES6 features.

### Possible Drawbacks
May not include all scenarios

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: _includes/markdown/JavaScript.md [hunk 1]
```diff
@@ -123,6 +123,80 @@ console.log( d ); // 4
 
 Use destructuring whenever possible to slim down your code and improve overall readability.
 
+#### Spread operator and Rest parameters
+The notation `...` which comes with ES6 represents the **spread** operator and **rest** parameters.
```
Comment (timwright12) [inline _includes/markdown/JavaScript.md:127]: @mserino can we focus this more so on the "why" and "when" for using the spread operator. Something like answering the question or "When is it a best practice to use this?" For code samples and such we can just link off site to MDN or something

**Review by timwright12 — CHANGES_REQUESTED** [review]

**Comment by timwright12** [comment]
Adding @nicholasio to this PR for review

**Comment by tlovett1** [comment]
Hey @mserino this is great! Should we also talk about objects (yes, I know arrays are technically objects but I think you know what I mean)?

**Comment by nicholasio** [comment]
@mserino Thanks for the PR - as mentioned before in this PR's comments. We want to keep the best practices more focused on "why"/"when" instead of getting more technical and teaching how to use a particular JS feature. With that said I'm going to close this one for now

**Pull request closed without merging**